common/src/main/java/com/wynntils/models/content/ContentBookQueries.java

```diff
@@ -0,0 +1,206 @@
+                    progress.add(InventoryUtils.getItemName(itemStack));
+                    progress.addAll(LoreUtils.getLore(itemStack));
+                })
+
+                // Repeatedly click next page, if available, and process the following page
+                .repeat(
+                        c -> ScriptedContainerQuery.containerHasSlot(
+                                c, NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, SCROLL_DOWN_TEXT),
+                        QueryStep.clickOnSlot(NEXT_PAGE_SLOT)
+                                .processIncomingContainer(c -> processContentBookPage(c, newContent)))
+
+                // Restore filter to original value
+                .execute(() -> filterLoopCount = 0)
+                .repeat(
+                        c -> {
+                            filterLoopCount++;
+                            if (filterLoopCount > MAX_FILTERS) {
+                                throw new ContainerQueryException("Filter setting has exceeded max loops");
+                            }
+
+                            String activeFilter = getActiveFilter(c.items().get(CHANGE_VIEW_SLOT));
+                            if (activeFilter == null) {
+                                throw new ContainerQueryException("Cannot determine active filter");
+                            }
+
+                            // Continue looping until filter matches original value
+                            return !activeFilter.equals(selectedFilter);
+                        },
+                        QueryStep.clickOnSlot(CHANGE_VIEW_SLOT))
+
+                // Finally signal we're done
+                .execute(() -> processResult.accept(newContent, progress))
+                .build();
+
+        query.executeQuery();
+    }
+
+    private String getActiveFilter(ItemStack itemStack) {
+        StyledText itemName = InventoryUtils.getItemName(itemStack);
+        if (!itemName.equals(StyledText.fromString("§eFilter"))) return null;
+
+        List<StyledText> lore = LoreUtils.getLore(itemStack);
+        for (StyledText line : lore) {
+            Matcher m = line.getMatcher(ACTIVE_FILTER);
+            if (m.matches()) {
+                return m.group(1);
+            }
+        }
+
+        return null;
+    }
+
+    private void processContentBookPage(ContainerContent container, List<ContentInfo> newContent) {
+        for (int slot = 0; slot < 54; slot++) {
+            ItemStack itemStack = container.items().get(slot);
+            Optional<ContentItem> contentItemOpt = Models.Item.asWynnItem(itemStack, ContentItem.class);
+            if (contentItemOpt.isEmpty()) continue;
+
+            ContentInfo contentInfo = contentItemOpt.get().getContentInfo();
+
+            newContent.add(contentInfo);
+        }
+    }
+
+    protected void toggleTracking(String name, ContentType contentType) {
+        // We do not want to change filtering when tracking, since we get
+        // no chance to reset it
```

Comment:
I am not sure why this is the case?

Reply:
As soon as you click an item to track it, the container closes, but any changes we have made to the filter state will remain.

Reply:
This I failed to observe, and added reset code to tracking... (although it is disabled atm)